scan.go

```diff
@@ -487,6 +500,33 @@ func internalStartScan() error {
 		enrichers = append(enrichers, enricher)
 	}
 
+	if enableCodeAnalysisEnricher() {
```

Comment:
We should first check if `code` analysis is enabled by providing a DB. Then create the storage, client etc. Then based on config flags create and append the enrichers.